utils/pattern/src/frontend/mod.rs

```diff
@@ -212,6 +212,36 @@ where
             store,
         })
     }
+
+    /// Creates a pattern from a UTF-8 encoded string.
+    ///
+    /// ✨ *Enabled with the `alloc` Cargo feature.*
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_pattern::Pattern;
+    /// use icu_pattern::PatternItemCow;
+    /// use icu_pattern::SinglePlaceholder;
+    /// use icu_pattern::SinglePlaceholderKey;
+    /// use std::borrow::Cow;
+    ///
+    /// Pattern::<SinglePlaceholder, _>::
+    ///                 try_from_utf8("{0} days".as_bytes())
+    ///                             .expect("single placeholder pattern");
+    /// ```
+    pub fn try_from_utf8(bytes: &[u8]) -> Result<Self, Error> {
+        let store = B::try_store_from_utf8(bytes).map_err(|_| Error::InvalidPattern)?;
+        #[cfg(debug_assertions)]
+        match B::validate_store(core::borrow::Borrow::borrow(&store)) {
+            Ok(()) => (),
+            Err(e) => {
+                debug_assert!(false, "Pattern validation failed: {:?}", e);
+            }
+        };
+
+        Ok(Self::from_store_unchecked(store.to_owned()))
+    }
```

Comment:
Issue: You should always call validate_store here, not just in debug_assertions

Reply:
done, but why this is not the case in `try_from_items` ?

Reply:
Because you are constructing from a store that you haven't validated yet

Reply:
but `try_store_from_utf8` has already validated it